frontend/src/app/filter/multiple-choice-filter/multiple-choice-filter.component.ts

```diff
@@ -29,17 +30,15 @@ export class MultipleChoiceFilterComponent extends BaseFilterComponent<MultipleC
     private async getOptions(): Promise<void> {
         if (this.filter && this.queryModel) {
             const optionCount = (this.filter.corpusField.filterOptions as MultipleChoiceFilterOptions).option_count;
-            const aggregator = { name: this.filter.corpusField.name, size: optionCount };
+            const aggregator = new TermsAggregator(this.filter.corpusField, optionCount);
             const queryModel = this.queryModel.clone();
             queryModel.filterForField(this.filter.corpusField).deactivate();
-            this.searchService.aggregateSearch(queryModel.corpus, queryModel, [aggregator]).then(
-                response => response.aggregations[this.filter.corpusField.name]).then(aggregations =>
-                    this.options = _.sortBy(
-                        aggregations.map(x => ({ label: x.key, value: x.key, doc_count: x.doc_count })),
-                        o => o.label
-                    )
-                ).catch(() => this.options = []);
-
+            this.searchService.aggregateSearch(queryModel.corpus, queryModel, aggregator).then(result =>
+                this.options = _.sortBy(
+                    result.map(x => ({ label: x.key, value: x.key, doc_count: x.doc_count })),
+                    o => o.label
+                )
+            ).catch(() => this.options = []);
```

Comment:
I would prefer meaningful variable names over `x` and `o`, or type annotations. The code is already quite difficult to read.

I know this is not in the scope of the PR, but just stood out to me now that it's presented to me :)